src/store/models/account.ts

```diff
@@ -58,25 +84,38 @@ export default createModel<RootModel>()({
       ...state,
       chosenTx,
     }),
+    SET_KEY: (state, key: string) => ({
+      ...state,
+      key,
+    }),
   },
   effects: dispatch => ({
     async login({ key, setError }: LoginArgs) {
       setError('');
-      nspvLogin(key)
+      const userKey = key || this.key;
```

Comment:
Not strictly necessary here since a valid key is not gonna be falsy, but I think we should always be preferring null coalescing operator since we're using it elsewhere.